src/main/java/seedu/address/model/property/FilterPropsByPricePredicate.java

```diff
@@ -0,0 +1,36 @@
+package seedu.address.model.property;
+
+import static java.util.Objects.requireNonNull;
+
+import seedu.address.model.pricerange.PriceRange;
+
+
+
+/**
+ * Tests that a {@code Property}'s {@code Price} is within the given price range.
+ */
+public class FilterPropsByPricePredicate extends AbstractFilterPropsPredicate {
+
+    private final PriceRange priceRange;
+
+    /**
+     * Standard constructor for the predicate.
+     */
+    public FilterPropsByPricePredicate(PriceRange priceRange) {
+        requireNonNull(priceRange);
+        this.priceRange = priceRange;
+    }
+
+    @Override
+    public boolean test(Property p) {
+        return priceRange.isWithinPriceRange(p.getPrice());
+    }
+
+    @Override
+    public boolean equals(Object other) {
+        return other == this // short circuit if same object
+                || (other instanceof FilterPropsByPricePredicate // instanceof handles nulls
+                && priceRange.equals(((FilterPropsByPricePredicate) other).priceRange)); // state check
+    }
```

Comment:
While looking at this method, I looked at `PriceRange::equals` as well, and realised it compares the `toString` results of both `PriceRange`. Would it be better to compare the high and low of the `PriceRange` instead?

Reply:
Ah, the reason why I decided to use the `toString()` method here is because that method contains the numerical values of `high` and `low` in a given price range. I can change it to check the double values directly if you think it's better, but there shouldn't be any major issues either way.